.github/workflows/multi-arch-build.yaml

```diff
@@ -65,69 +72,69 @@ jobs:
           test -n "$VERSION"
           echo "::set-output name=version::${VERSION}"
 
-      # Generate image FQINs, labels, check whether to push
-      - name: Generate image information
-        id: image_info
+      - name: Generate podman reg. image FQIN(s)
+        id: podman_reg
         run: |
-          VERSION='v${{ steps.sniff_test.outputs.version }}'
-          # workaround vim syntax-hilighting bug: '
           if [[ "${{ matrix.source }}" == 'stable' ]]; then
-            # quay.io/podman/stable:vX.X.X
+            # The `podman version` in image just built
+            VERSION='v${{ steps.sniff_test.outputs.version }}'
+            # workaround vim syntax-highlight bug: '
+            # Image tags previously pushed to quay
             ALLTAGS=$(skopeo list-tags \
                       docker://$PODMAN_QUAY_REGISTRY/stable | \
                       jq -r '.Tags[]')
-            PUSH="false"
-            if ! fgrep -qx "$VERSION" <<<"$ALLTAGS"; then
-                PUSH="true"
-            fi
 
-            FQIN="$PODMAN_QUAY_REGISTRY/stable:$VERSION"
-            # Only push if version tag does not exist
-            if [[ "$PUSH" == "true" ]]; then
-              echo "Will push $FQIN"
-              echo "::set-output name=podman_push::true"
-              echo "::set-output name=podman_fqin::${FQIN}"
-            else
-              echo "Not pushing, $FQIN already exists."
+            # New image? Push quay.io/podman/stable:vX.X.X and :latest
+            if ! fgrep -qx "$VERSION" <<<"$ALLTAGS"; then
+              # Assume version-tag is also the most up to date (i.e. "latest")
+              FQIN="$PODMAN_QUAY_REGISTRY/stable:$VERSION,$PODMAN_QUAY_REGISTRY/stable:latest"
+            else # Not a new version-tagged image
+              # Assume other contents changed, so this is the "new" latest.
+              FQIN="$PODMAN_QUAY_REGISTRY/stable:latest"
```

Comment:
This is what bothers me. As a naïve end user, I do not want "other contents changed". I do not want `stable:latest` to be some unpredictable unknown random thing that differs from `stable:<most-recent-tag>`. I want `stable` to be... stable.

I realize I'm arriving late into this game, and may not have full context of what these images are used for, and there may be something obvious I'm missing. But this really seems like a huge deal to me, and I need to understand our intention before I can continue reviewing.

Reply:
Oh I see what you mean now.  So to answer (maybe) the most important question (@TomSweeneyRedHat please correct me):

> what these images are used for

Nothing and (probably) nobody uses them.  How do we know this?  There was a serious problem in the prior implementation, where the `podman` image was actually built using the buildah repo. (yep, you read that correctly).  This went on for months and months, and we got nary a single complaint :grin: 

This github action is a royal PITA to debug.  Look, this is the 3rd PR just to get it kind of working properly.  It can only actually execute on master :confounded:   Adding any more script newfangles at this point scares me as a months-long endeavor :cry: 

So here's what I propose:

Let's get this (imperfect) PR in, and any more required to make things sort-of functional.  Then I need to think about fixing buildah, and skopeo which both already has a substantially similar (and worse broken) implementations.  Part of that journey should include simplifying a lot of this script-in-YAML into shared code it's not duplicated 3x.  That then would be a fantastical place to sneak in tweaks for the proper handling of "latest" as you describe.

Sound reasonable?

Reply:
The stable version of this container image is just built with the released version of Podman for the current Fedora.

Testing would be just the image that is in updates-testing, which means most of the time they are the same.

upstream would have the latest podman built from the upstream bits.

Reply:
That's Correct @rhatdan .  The issue @edsantiago raised is regarding the "latest" tagged image.  He believes they should match the most recent "version-tagged" image.  As-written currently, the "latest" images are built and pushed daily.

[For example, in this most recent successful run]( https://github.com/containers/podman/actions/runs/808065332), The `quay.io/podman/stable:latest` image is now F34 w/ podman v3.1.2 (and will update daily).  Whereas the most recent "version-tagged" image, `quay.io/podman/stable:v3.1.2` is using F33 (and will never be updated).

I'm avoiding "fixing" this for now, because the automation is already very complicated and difficult to debug.  I'm also afraid of several important corner-cases that need handling.  Mainly, we would never get a "new" latest-tag, despite upstream Dockerfile changes or new OS security updates, unless/until a new podman version is released - which might be weeks.  

So, we agreed to merged this as-is (w/ daily updates/pushing of "latest") and I agreed to address the issue separately.

Reply:
I opened a jira card to track the "fix this later" intention.